parsing/location.ml

```diff
@@ -423,10 +438,8 @@ let () =
       | Sys_error msg ->
           Some (errorf ~loc:(in_file !input_name)
                 "I/O error: %s" msg)
-      | Warnings.Errors n ->
-          Some
-            (errorf ~loc:(in_file !input_name)
-             "Some fatal warnings were triggered (%d occurrences)" n)
+
+      | Warnings.Errors _ -> Some non_displayed_error
```

Comment:
I'd rather have the compiler behave more like a library and less like a standalone program, so I don't really want to add calls to exit.
I don't see how the refactoring you suggest would help with reporting multiple type errors: it sounds like reporting such errors the same way as warning would work fine.

If the comparison with the empty string is what you want to remove, maybe renaming `Warnings.Errors` to `Location.Already_displayed` and changing `Location.error_of_exn` to have type `exn -> [ Ok of error | Already_displayed ] option` would work. This way `Ast_mapper.ext_of_exn` can serialize the Already_displayed case, and rethrow on the other side, and `Location.report_exception` would behave as you wrote earlier.